Missing docs for making use of TypeScript DSL API

### Problem

PR #658 added

> a TypeScript definition file for the DSL API, complete with documentation comments for IntelliSense.

Since the term "IntelliSense" is specific to Visual Studio, I'm assuming that these are intended to to at least be usable in Visual Studio Code (and probably also in other editors I'm less familiar with). The [Creating parsers](https://tree-sitter.github.io/tree-sitter/creating-parsers) docs mention that the `grammar.js` file generated by `tree-sitter init` should include these lines, which cause editors to look for that `dsl.d.ts` file:

```js
/// <reference types="tree-sitter-cli/dsl" />
// @ts-check
```

As of commit b60b2489aa1324b9efa0966bff21b779ca702537 by @ObserverOfTime and @amaanq, this file is not provided in the template used by `tree-sitter init`, so no editor can find it by default. However, the `@ts-check` line still causes the editor to treat the file as TypeScript, causing errors; for example, VS Code gives the following two errors for the default `grammar.js` file:

- Cannot find type definition file for 'tree-sitter-cli/dsl'.
- Cannot find name 'grammar'.


I was able to get rid of those errors by running `npm install` which is not mentioned in the docs, but on my machine (I use [nvm](https://github.com/nvm-sh/nvm) and [pyenv](https://github.com/pyenv/pyenv), and currently have Node v18.20.3 and Python v3.12.3 selected as my defaults) it fails with this error:

```
$ npm install        

> tree-sitter-carbon@0.1.0 install
> node-gyp-build

gyp info it worked if it ends with ok
gyp info using node-gyp@10.1.0
gyp info using node@18.20.3 | darwin | arm64
gyp info find Python using Python version 3.12.3 found at "/Users/samueles/.pyenv/versions/3.12.3/bin/python3"

gyp info spawn /Users/samueles/.pyenv/versions/3.12.3/bin/python3
gyp info spawn args [
gyp info spawn args '/Users/samueles/.nvm/versions/node/v18.20.3/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args 'binding.gyp',
gyp info spawn args '-f',
gyp info spawn args 'make',
gyp info spawn args '-I',
gyp info spawn args '/Users/samueles/github/samestep/tree-sitter-carbon/build/config.gypi',
gyp info spawn args '-I',
gyp info spawn args '/Users/samueles/.nvm/versions/node/v18.20.3/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args '-I',
gyp info spawn args '/Users/samueles/Library/Caches/node-gyp/18.20.3/include/node/common.gypi',
gyp info spawn args '-Dlibrary=shared_library',
gyp info spawn args '-Dvisibility=default',
gyp info spawn args '-Dnode_root_dir=/Users/samueles/Library/Caches/node-gyp/18.20.3',
gyp info spawn args '-Dnode_gyp_dir=/Users/samueles/.nvm/versions/node/v18.20.3/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args '-Dnode_lib_file=/Users/samueles/Library/Caches/node-gyp/18.20.3/<(target_arch)/node.lib',
gyp info spawn args '-Dmodule_root_dir=/Users/samueles/github/samestep/tree-sitter-carbon',
gyp info spawn args '-Dnode_engine=v8',
gyp info spawn args '--depth=.',
gyp info spawn args '--no-parallel',
gyp info spawn args '--generator-output',
gyp info spawn args 'build',
gyp info spawn args '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  TOUCH Release/obj.target/node_modules/node-addon-api/node_addon_api_except.stamp
  CXX(target) Release/obj.target/tree_sitter_carbon_binding/bindings/node/binding.o
In file included from ../bindings/node/binding.cc:1:
../node_modules/node-addon-api/napi.h:14:10: fatal error: 'functional' file not found
   14 | #include <functional>
      |          ^~~~~~~~~~~~
1 error generated.
make: *** [Release/obj.target/tree_sitter_carbon_binding/bindings/node/binding.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack at ChildProcess.<anonymous> (/Users/samueles/.nvm/versions/node/v18.20.3/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:209:23)
gyp ERR! System Darwin 24.0.0
gyp ERR! command "/Users/samueles/.nvm/versions/node/v18.20.3/bin/node" "/Users/samueles/.nvm/versions/node/v18.20.3/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/samueles/github/samestep/tree-sitter-carbon
gyp ERR! node -v v18.20.3
gyp ERR! node-gyp -v v10.1.0
gyp ERR! not ok 
npm error code 1
npm error path /Users/samueles/github/samestep/tree-sitter-carbon
npm error command failed
npm error command sh -c node-gyp-build

npm error A complete log of this run can be found in: /Users/samueles/.npm/_logs/2024-10-30T17_55_06_490Z-debug-0.log
```

### Steps to reproduce

Install the Tree-sitter CLI (specifically I installed it via Cargo) and run `tree-sitter init`; the specific values provided don't matter:

```
$ tree-sitter init
✔ Parser name · foo
✔ CamelCase name · Foo
✔ Description · Foo grammar for tree-sitter
✔ Repository URL · https://example.com
✔ TextMate scope · source.foo
✔ File types (space-separated) · .foo
✔ Version · 0.1.0
✔ License · MIT
✔ Author name · Foo
✔ Author email · 
✔ Author URL · 
Your current configuration:
{
  "name": "foo",
  "camelcase": "Foo",
  "description": "Foo grammar for tree-sitter",
  "repository": "https://example.com/",
  "scope": "source.foo",
  "file_types": [
    ".foo"
  ],
  "version": "0.1.0",
  "license": "MIT",
  "author": "Foo"
}
✔ Does the config above look correct? · yes
```

Then open the generated `grammar.js` file in an editor that provides IntelliSense, and observe the errors that appear.

### Expected behavior

I expected the docs page to mention that `npm install` needs to be run in order to get rid of the error messages in editors that support IntelliSense. I also expected that `npm install` would succeed. If a specific version for Node and/or Python are required, I would expect those to be listed in the docs.

### Tree-sitter version (tree-sitter --version)

tree-sitter 0.24.3

### Operating system/version

macOS Sequoia 15.0.1

Update: I followed https://github.com/nodejs/node-addon-api/issues/222#issuecomment-2413550446 to reinstall my Xcode Command Line Tools, and now I get a different error:

```
$ npm install

> tree-sitter-carbon@0.1.0 install
> node-gyp-build

gyp info it worked if it ends with ok
gyp info using node-gyp@10.1.0
gyp info using node@18.20.3 | darwin | arm64
gyp info find Python using Python version 3.12.3 found at "/Users/samueles/.pyenv/versions/3.12.3/bin/python3"

gyp info spawn /Users/samueles/.pyenv/versions/3.12.3/bin/python3
gyp info spawn args [
gyp info spawn args '/Users/samueles/.nvm/versions/node/v18.20.3/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args 'binding.gyp',
gyp info spawn args '-f',
gyp info spawn args 'make',
gyp info spawn args '-I',
gyp info spawn args '/Users/samueles/github/samestep/tree-sitter-carbon/build/config.gypi',
gyp info spawn args '-I',
gyp info spawn args '/Users/samueles/.nvm/versions/node/v18.20.3/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args '-I',
gyp info spawn args '/Users/samueles/Library/Caches/node-gyp/18.20.3/include/node/common.gypi',
gyp info spawn args '-Dlibrary=shared_library',
gyp info spawn args '-Dvisibility=default',
gyp info spawn args '-Dnode_root_dir=/Users/samueles/Library/Caches/node-gyp/18.20.3',
gyp info spawn args '-Dnode_gyp_dir=/Users/samueles/.nvm/versions/node/v18.20.3/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args '-Dnode_lib_file=/Users/samueles/Library/Caches/node-gyp/18.20.3/<(target_arch)/node.lib',
gyp info spawn args '-Dmodule_root_dir=/Users/samueles/github/samestep/tree-sitter-carbon',
gyp info spawn args '-Dnode_engine=v8',
gyp info spawn args '--depth=.',
gyp info spawn args '--no-parallel',
gyp info spawn args '--generator-output',
gyp info spawn args 'build',
gyp info spawn args '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  TOUCH Release/obj.target/node_modules/node-addon-api/node_addon_api_except.stamp
  CXX(target) Release/obj.target/tree_sitter_carbon_binding/bindings/node/binding.o
  SOLINK_MODULE(target) Release/tree_sitter_carbon_binding.node
clang++: error: no such file or directory: 'Release/obj.target/tree_sitter_carbon_binding/src/parser.o'
make: *** [Release/tree_sitter_carbon_binding.node] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack at ChildProcess.<anonymous> (/Users/samueles/.nvm/versions/node/v18.20.3/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:209:23)
gyp ERR! System Darwin 24.0.0
gyp ERR! command "/Users/samueles/.nvm/versions/node/v18.20.3/bin/node" "/Users/samueles/.nvm/versions/node/v18.20.3/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/samueles/github/samestep/tree-sitter-carbon
gyp ERR! node -v v18.20.3
gyp ERR! node-gyp -v v10.1.0
gyp ERR! not ok 
npm error code 1
npm error path /Users/samueles/github/samestep/tree-sitter-carbon
npm error command failed
npm error command sh -c node-gyp-build

npm error A complete log of this run can be found in: /Users/samueles/.npm/_logs/2024-10-30T20_03_50_154Z-debug-0.log
```

--- a/docs/section-3-creating-parsers.md
+++ b/docs/section-3-creating-parsers.md
@@ -86,6 +86,12 @@ This should print the following:
 
 You now have a working parser.
 
+Finally, look back at the [triple-slash][] and [`@ts-check`][ts-check] comments in `grammar.js`; these tell your editor to provide documentation and type information as you edit your grammar. For these to work, you must download Tree-sitter's TypeScript API from npm into a `node_modules` directory in your project:
+
+```sh
+npm install
+```
+
 ## Tool Overview
 
 Let's go over all of the functionality of the `tree-sitter` command line tool.

